# Gauchian-enrich

[![DOI](https://img.shields.io/badge/DOI-10.5281%2Fzenodo.16366908-blue?style=flat-square)](https://doi.org/10.5281/zenodo.16366908)&nbsp;&nbsp;
[![Gauchian-enrich (RRID:SCR_027296)](https://img.shields.io/badge/RRID-Gauchian--enrich%20(RRID%3ASCR_027296)-blue?style=flat-square)](https://scicrunch.org/resolver/RRID:SCR_027296)

A variant annotation toolkit for the GBA gene variants detected from the [Illumina/Gauchian](https://github.com/Illumina/Gauchian) tool.

## Key Features

- **Transforms basic variant calls**: Converts raw output from Illumina Gauchian into richly annotated, interpretation-ready variants.  It provides detailed biological context for genetic variants, including transcript information, precise genomic coordinates (gDNA/cDNA/protein), and a clear assessment of their functional impact.
- **Detailed Functional Analysis**: Goes beyond basic variant calls to classify mutations (e.g., missense) and suggests potential alternative causes, such as multiple nucleotide variants (MNVs), that might lead to the same protein change.
- **Rich Output Format**: The tool generates a highly detailed, tabular output designed for seamless integration into existing bioinformatics pipelines. The output includes essential fields like `CHROM`, `POS`, `REF`, and `ALT`, making it easily parseable and ideal for downstream analysis.

## Setup (recommended)

To use the `gauchian_enrich` utility, ensure Docker is installed on your system. Then, pull the Docker image:

```bash
docker pull ghcr.io/maurya-anand/gauchian-enrich
```

## Manual setup

If you prefer to install and configure the `gauchian_enrich` utility locally without using Docker, you can use the provided `Makefile`. This will ensure all dependencies are installed and the required configuration files are set up.

### Steps for Manual Installation

1. Ensure the following tools are installed on your system:
   - `samtools`
   - `wget`
   - `gunzip`
   - `pip` or `pip3`
   - Perl module `Parallel::ForkManager`

2. Run the following command to install and configure the utility:

```bash
make install
```

## Usage

**Step 1** : Run the [Illumina/Gauchian](https://github.com/Illumina/Gauchian) tool to obtain the `.tsv` and `.json` output.

**Step 2** : Use the `.tsv` ouput file from the previous step as an input to `gauchian_enrich` utility.

### Command-Line Options

- `--input` or `-i`: Specify the input `.tsv` file generated by the Gauchian tool.
- `--output` or `-o`: Specify the output file path. If omitted, a default output file will be created based on the input file name.
- `--help` or `-h`: Display usage instructions.
- `--version` or `-v`: Display the version of the `gauchian_enrich` tool.

>[!Note]
> The `gauchian_enrich` utility performs reverse annotation using the `transvar` tool and the `hg38` reference genome.
> The reverse annotation process relies on the `hg38` reference genome file (`/ref/hg38.fa`). The Docker image for `gauchian_enrich` comes preconfigured with `transvar` and the `hg38` reference genome, eliminating the need for users to download or configure the reference genome manually.

### Example usage

To display usage instructions:

```bash
docker run ghcr.io/maurya-anand/gauchian-enrich gauchian_enrich --help
```

To execute the tool using the given example data:

```bash
docker run -u $(id -u):$(id -g) -it \
    -v ${PWD}/example:/example \
    ghcr.io/maurya-anand/gauchian-enrich \
    gauchian_enrich -i /example/test.gauchian.tsv -o /example/test.gauchian.annotated.tsv
```

## Example data

### Input

| Sample           | is_biallelic(GBAP1-like_variant_exon9-11) | is_carrier(GBAP1-like_variant_exon9-11) | CN(GBA+GBAP1) | deletion_breakpoint_in_GBA | GBAP1-like_variant_exon9-11 | other_unphased_variants |
|-------------------|------------------------------------------|------------------------------------------|---------------|-----------------------------|-----------------------------|--------------------------|
| NA20815.final     | FALSE                                    | TRUE                                     | 4             | N/A                         | L483P/

### Output

| Sample         | is_biallelic(GBAP1-like_variant_exon9-11) | is_carrier(GBAP1-like_variant_exon9-11) | CN(GBA+GBAP1) | deletion_breakpoint_in_GBA | GBAP1-like_variant_exon9-11 | other_unphased_variants | input      | transcript                    | gene | strand | coordinates(gDNA/cDNA/protein)                          | region                   | info                                                                                                                                                                                                                                                                                                                      | CHROM | POS       | REF | ALT |
|----------------|--------------------------------------------|------------------------------------------|----------------|-----------------------------|------------------------------|--------------------------|------------|----------------------------------|------|--------|---------------------------------------------------------|--------------------------|---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|--------|------------|-----|-----|
| NA20815.final  | FALSE                                      | TRUE                                     | 4              | N/A                         | L483P/                       | None                     | GBA:L483P | NM_001005742.2 (protein_coding) | GBA  | -      | chr1:g.155235252A>G/c.1448T>C/p.L483P                  | inside_[cds_in_exon_11] | CSQN=Missense;reference_codon=CTG;candidate_codons=CCT,CCG,CCA,CCC;candidate_mnv_variants=chr1:g.155235251_155235252delCAinsAG,chr1:g.155235251_155235252delCAinsTG,chr1:g.155235251_155235252delCAinsGG;dbxref=GeneID:2629,HGNC:HGNC:4177,MIM:606463;aliases=NP_001005742;source=RefSeq | chr1   | 155235252 | A   | G   |

## Reference

- Toffoli, M., Chen, X., Sedlazeck, F.J. et al. Comprehensive short and long read sequencing analysis for the Gaucher and Parkinson’s disease-associated GBA gene. Commun Biol 5, 670 (2022). [https://doi.org/10.1038/s42003-022-03610-7](https://doi.org/10.1038/s42003-022-03610-7)
- Zhou, W., Chen, T., Chong, Z. et al. TransVar: a multilevel variant annotator for precision genomics. Nat Methods 12, 1002–1003 (2015). [https://doi.org/10.1038/nmeth.3622](https://doi.org/10.1038/nmeth.3622)

## Citations

If you use this tool for your analysis, please cite the `gauchian-enrich` as follows:

> SciCrunch ID: [Gauchian-enrich (RRID:SCR_027296)](https://scicrunch.org/resolver/RRID:SCR_027296)
>
> Anand Maurya. (2025). maurya-anand/gauchian-enrich: v1.0.0 (v1.0.0). Zenodo. [https://doi.org/10.5281/zenodo.16366908](https://doi.org/10.5281/zenodo.16366908)
